pkg/cmd/server/api/v1/conversions.go

```diff
@@ -17,6 +17,15 @@ func addDefaultingFuncs(scheme *runtime.Scheme) {
 			if len(obj.APILevels) == 0 {
 				obj.APILevels = internal.DefaultOpenShiftAPILevels
 			}
+			if len(obj.JenkinsPipelineConfig.Namespace) == 0 {
+				obj.JenkinsPipelineConfig.Namespace = "openshift"
+			}
+			if len(obj.JenkinsPipelineConfig.TemplateName) == 0 {
+				obj.JenkinsPipelineConfig.TemplateName = "jenkins"
+			}
+			if len(obj.JenkinsPipelineConfig.ServiceName) == 0 {
+				obj.JenkinsPipelineConfig.ServiceName = "jenkins-ci"
```

Comment:
This PR is implementing both 1 and 2

On Wed, Mar 30, 2016 at 4:27 PM, Ben Parees <notifications@github.com>
wrote:

> In pkg/cmd/server/api/v1/conversions.go
> <https://github.com/jimmidyson/origin/pull/2#discussion_r57957777>:
>
> > @@ -17,6 +17,15 @@ func addDefaultingFuncs(scheme *runtime.Scheme) {
> >  			if len(obj.APILevels) == 0 {
> >  				obj.APILevels = internal.DefaultOpenShiftAPILevels
> >  			}
> > +			if len(obj.JenkinsPipelineConfig.Namespace) == 0 {
> > +				obj.JenkinsPipelineConfig.Namespace = "openshift"
> > +			}
> > +			if len(obj.JenkinsPipelineConfig.TemplateName) == 0 {
> > +				obj.JenkinsPipelineConfig.TemplateName = "jenkins"
> > +			}
> > +			if len(obj.JenkinsPipelineConfig.ServiceName) == 0 {
> > +				obj.JenkinsPipelineConfig.ServiceName = "jenkins-ci"
>
> let's remember that there are two scenarios we are targeting:
>
> 1) jenkinsfile pipelines executed by jenkins
> 2) "some pipeline definition" executed by "some pipeline engine"
>
> this PR is targeted at (1) so it's appropriate to include jenkins in the
> names of the things it creates/relies on.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/jimmidyson/origin/pull/2/files/3ed08d495dd391ab47317d2946c2695da23b317b#r57957777>
>